iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m

```diff
@@ -197,16 +199,23 @@ + (void)sendCurrentAuthStatusToListeners {
     [self sendAndClearLocationListener:!denied];
 }
 
-+ (void)internalGetLocation:(bool)prompt {
++ (void)internalGetLocation:(bool)prompt fallbackToSettings:(BOOL)fallback {
+    fallbackToSettings = fallback;
+    id clLocationManagerClass = NSClassFromString(@"CLLocationManager");
+    
     if ([self started]) {
-        [self sendCurrentAuthStatusToListeners];
+        CLAuthorizationStatus permissionStatus = [clLocationManagerClass performSelector:@selector(authorizationStatus)];
+
+        if (prompt && fallback && permissionStatus == kCLAuthorizationStatusDenied)
```

Comment:
add a comment for what these 3 checks indicate:
1. prompt...
2. fallback...
3. permissionStatus...

I can see myself looking at this some day confused haha